google/generativeai/types/content_types.py

```diff
@@ -157,17 +160,53 @@ def to_blob(blob: BlobType) -> glm.Blob:
         )
 
 
+class FileDataDict(TypedDict):
+    mime_type: str
+    file_uri: str
+
+
+FileDataType = Union[FileDataDict, glm.FileData, file_types.File]
+
+
+def to_file_data(file_data: FileDataType):
+    if isinstance(file_data, dict):
+        if "file_uri" in file_data:
+            file_data = glm.FileData(file_data)
+        else:
+            file_data = glm.File(file_data)
+
+    if isinstance(file_data, file_types.File):
+        file_data = file_data.to_proto()
+
+    if isinstance(file_data, (glm.File, file_types.File)):
+        file_data = glm.FileData(
+            mime_type=file_data.mime_type,
+            file_uri="https://generativelanguage.googleapis.com/" + file_data.name,
```

Comment:
file.uri  should be ready to be used so we don't need the concatenation anymore

Reply:
Works. Done.